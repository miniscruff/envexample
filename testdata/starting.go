package testdata

// StartingConfig handles a configuration for a project with default env options.
type StartingConfig struct {
	// NoEnvStructTag at all
	NoEnvStructTag string
	// NoEnvName where env struct tag is empty
	NoEnvName string `env:""`
	// WithEnvName sets our env var key
	WithEnvName string `env:"WITH_ENV_NAME"`
	// RequiredNoDefault specifies a required value but no default
	RequiredNoDefault string `env:",required"`
	// NotEmpty specifies a field is required and can not be empty
	NotEmpty string `env:",notEmpty,,"`
	// FromFile loads our value from a file at path
	FromFile string `env:",file"`
	// UnsetVar will unset our environment variable after it is loaded
	UnsetVar string `env:",unset"`

	// Not part of our .env.example
	unexportedIgnored string `env:"ignored"`
}

/*


	Home         string        `env:"HOME"`
	Port         int           `env:"PORT" envDefault:"3000"`
	Password     string        `env:"PASSWORD,unset"`
	IsProduction bool          `env:"PRODUCTION"`
	Hosts        []string      `env:"HOSTS" envSeparator:":"`
	Duration     time.Duration `env:"DURATION"`
	TempFolder   string        `env:"TEMP_FOLDER" envDefault:"${HOME}/tmp" envExpand:"true"`
*/
