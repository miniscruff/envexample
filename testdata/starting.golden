# Generated by envexample dev.

# StartingConfig handles a configuration for a project with default env options.
#
# FromFile loads our value from a file at path
# Info: Value is loaded from a file at the path defined.
#FromFile=""
# NoEnvName where env struct tag is empty
#NoEnvName=""
# NoEnvStructTag at all
#NoEnvStructTag=""
# NotEmpty specifies a field is required and can not be empty
NotEmpty="can not be empty"
# RequiredNoDefault specifies a required value but no default
RequiredNoDefault=""
# UnsetVar will unset our environment variable after it is loaded
# Info: After loading environment variable is unset.
#UnsetVar=""
# WithEnvName sets our env var key
#WITH_ENV_NAME=""
